File: backend-v2/src/main/java/vn/io/vutiendat3601/shop/v2/coupon/CouponDto.java
package vn.io.vutiendat3601.shop.v2.coupon;

import java.math.BigDecimal;
import java.time.ZonedDateTime;

public record CouponDto(
    String code,
    String name,
    String description,
    double discountRatio,
    BigDecimal maxAmount,
    ZonedDateTime expiredAt,
    String categoryCode,
    String productNo,
    String customerCode,
    CouponType type,
    CouponObjectType objectType) {}

File: frontend-v2/src/app/domain/coupon/coupon-dto.ts
export class CouponDto {
  constructor(
    public code: string,
    public name: string,
    public description: string,
    public discountRatio: number,
    public maxAmount: number,
    public expiredAt: string,
    public categoryCode: string,
    public productNo: string,
    public customerCode: string,
    public type: string,
    public objectType: string
  ) {}
}
